src/Cli/dotnet/commands/dotnet-run/RunCommand.cs

```diff
@@ -201,77 +212,128 @@ private List<string> GetRestoreArguments()
 
             // --interactive need to output guide for auth. It cannot be
             // completely "quiet"
-            if (!RestoreArgs.Any(a => a.StartsWith("-verbosity:")))
+            if (!cliRestoreArgs.Any(a => a.StartsWith("-verbosity:")))
             {
                 var defaultVerbosity = Interactive ? "minimal" : "quiet";
                 args.Add($"-verbosity:{defaultVerbosity}");
             }
 
-            args.AddRange(RestoreArgs);
+            args.AddRange(cliRestoreArgs);
 
-            return args;
+            return args.ToArray();
         }
 
         private ICommand GetTargetCommand()
         {
-            var globalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
-            {
-                // This property disables default item globbing to improve performance
-                // This should be safe because we are not evaluating items, only properties
-                { Constants.EnableDefaultItems,    "false" },
-                { Constants.MSBuildExtensionsPath, AppContext.BaseDirectory }
-            };
+            // TODO for MSBuild usage here: need to sync loggers (primarily binlog) used with this evaluation
+            var project = EvaluateProject(ProjectFileFullPath, RestoreArgs);
```

Comment:
Extracted prior MSBuild evaluation to a method for legibility, then made each of the following 'transformation' steps methods as well so it all flows.